### PR TITLE
Values structure refinement

### PR DESCRIPTION
**Pull request by lazyoldbear**
Major refinement of the `values.yaml` file.
Addresses:
* SERVER-540
* SERVER-1183
* SERVER-908

### CODE REVIEW

File: charts/document-engine/values.simple.yaml [hunk 1]
```diff
@@ -1,43 +1,48 @@
...
+
+documentSigningService:
+  enabled: true
+  url: http://signing-service:6000/sign
+  timeoutSeconds: 10
+
+assetStorage:
+  databaseMigrationJob: 
+    enabled: false
+  databaseEngine: postgresql
```
Comment (tomassurin) [inline charts/document-engine/values.simple.yaml:21]: We always need DB engine if you work with DE in "Server/with storage mode", it does not necessarily relate to asset storage. What about splitting it? e.g.:

```
# Config related to the main DE database.
database: 
  databaseMigrationJob: 
  enabled: false
  databaseEngine: postgres
  # I prefer the same key here as in the `databaseEngine` above
  postgres: 
     ...

# Config related to asset storage.
assetStorage:
   assetSotrageBackend: built-in
   enableAssetStorageFallback: false
```

File: charts/document-engine/values.yaml [hunk 2]
```diff
@@ -10,91 +10,79 @@ image:
...
-        passwordKey: DASHBOARD_PASSWORD
+documentEngineLicense:
+  # Activation key for online activation (most common) or
+  # license key for offline activation.
+  # Results in `ACTIVATION_KEY` environment variable
+  # or `LICENSE_KEY` (if `isOffline` is `true`)
+  activationKey: ""
+  # Is the license offline: set to `true` if
+  # an offline license key was provided in `activationKey`
+  isOffline: false
```
Comment (tomassurin) [inline charts/document-engine/values.yaml:26]: You don't need this anymore, you can use `ACTIVATION_KEY` at all times since 1.2.0 https://pspdfkit.com/changelog/document-engine/#1.2.0

File: charts/document-engine/values.yaml [hunk 3]
```diff
@@ -10,91 +10,79 @@ image:
...
+    jwtAlgorithmKey: JWT_ALGORITHM
+    # If external secret is enabled, but `jwtPublicKeyKey` is not set,
+    # the public key will be retrieved from the `apiAuth.jwt.publicKey` value
+    jwtPublicKeyKey: JWT_PUBLIC_KEY
+    # If external secret is enabled, but `secretKeyBaseKey` is not set,
+    # the secret key base will be retrieved from the `apiAuth.secretKeyBase` value
+    # or generated automatically
+    secretKeyBaseKey: SECRET_KEY_BASE
+
+config:
```
Comment (tomassurin) [inline charts/document-engine/values.yaml:69]: This is a bit strange top level name. What kind of "config" is this section related to? Looks like DE misc configuration.

Example split into more logical sections: 

```
coreWorker:
   poolSize: 16
   readAnnotationBatchTimeoutSeconds: 20
   # Technically it's handled by core, but don't really like to have it here. Maybe something where we group which features are available (other candidates for that are in `limits`)?
   automaticLinkExtraction: true
   
documentEngineTimeouts:
   requestTimeoutSeconds: 60
   workerTimeoutSeconds: 60
   generationTimeoutSeconds: 60
   urlFetchTimeoutSeconds: 5

limits:
    maxUploadSizeMegaBytes: 950
    allowDocumentUploads: true
    allowRemoteDocuments: true
    # Maybe these could be even grouped under generation?
    allowDocumentGeneration: true
    allowRemoteAssetsInGeneration: true

asyncJobs: 
   jobsTtlSeconds: 172800
   # Possibly stuff like `backend: built-in` if we want to make the jobs implementation more backend agnostic (e.g. not require Oban in Postgres only).

network: 
   proxy: 
      http: ""
      https: ""

replaceSecretsFromEnv: true
```

File: charts/document-engine/values.simple.yaml (hunk 1 above)
Comment (lazyoldbear) [reply charts/document-engine/values.simple.yaml:21]: This is exactly what I thought about. Hesitated. Can still do.

File: charts/document-engine/values.yaml (hunk 2 above)
Comment (lazyoldbear) [reply charts/document-engine/values.yaml]: Addressed:
* https://github.com/PSPDFKit/helm-charts/pull/20/commits/5eb931808a53e3455f781a5911f8a7a23074ece3
* https://github.com/PSPDFKit/PSPDFKit-Website/pull/10352/commits/c2e735196fa66f0d3c843997f0b41771c77a292a
